src/channels/components/ChannelDeleteDialog/ChannelDeleteDialog.tsx

```diff
@@ -42,52 +44,56 @@ const ChannelDeleteDialog: React.FC<ChannelDeleteDialogProps> = ({
       confirmButtonState={confirmButtonState}
       open={open}
       onClose={onClose}
-      onConfirm={() => (hasChannels ? onConfirm(choice) : onBack())}
+      onConfirm={() =>
+        hasChannels || !hasOrders ? onConfirm(choice) : onBack()
```

Comment:
It seems to be reused multiple times. How about :
const canBeDeleted = hasChannels || !hasOrders